iroh-gateway/src/config.rs

```diff
@@ -109,11 +111,26 @@ fn default_headers() -> HeaderMap {
             ACCEPT,
             CACHE_CONTROL,
             RANGE,
+            CONTENT_TYPE,
             HEADER_SERVICE_WORKER.clone(),
+            HEADER_X_REQUESTED_WITH.clone(),
+            USER_AGENT,
         ]
         .into_iter()
         .collect::<AccessControlAllowHeaders>(),
     );
+    headers.typed_insert(
+        [
+            CONTENT_LENGTH,
+            CONTENT_RANGE,
+            HEADER_X_IPFS_PATH.clone(),
+            HEADER_X_IPFS_ROOTS.clone(),
+            HEADER_X_CHUNKED_OUTPUT.clone(),
+            HEADER_X_STREAM_OUTPUT.clone(),
+        ]
+        .into_iter()
+        .collect::<AccessControlExposeHeaders>(),
+    );
```

Comment:
Note that the CORS headers can be set in the config file, which is a more flexible solution than hardcoding them here. I agree sane defaults are good, but what qualifies as sane with CORS? (lol)

Reply:
I will have university degree in this CORS crap soon, access-headers chase me in nightmares

There are two suggestions to define sanity. The first one is passing tests for base web-cases. Having nice defaults for then is not just important but vital. I have experience in distributing IPFS-based apps armed with instruction how to install it and people ALWAYS had been complaining about configuration part. Going to terminal to copy and paste some configuration lines are too hard for most people.

For example, without headers from this PR I wasn't able to access server response from inside service workers. IMO, it is essential because caching inside service workers are important for dWeb-apps, without caching app will degrade severely. 

Another option is just copy headers from kubo. They walked this path already and I believe CORS part may be taken from kubo without brainstorming because deep and sane analysis for CORS will fail anyway lol.

Reply:
I agree having some sane defaults is good, will defer to @Arqu and @b5 for the actual values.